unstructured/partition/xlsx.py

```diff
@@ -85,26 +92,43 @@ def partition_xlsx(
     last_modification_date = None
     header = 0 if include_header else None
 
+    sheets: dict[str, pd.DataFrame] = {}
     if filename:
-        sheets = pd.read_excel(filename, sheet_name=None, header=header)
+        sheets = pd.read_excel(  # pyright: ignore[reportUnknownMemberType]
+            filename, sheet_name=None, header=header
+        )
         last_modification_date = get_last_modified_date(filename)
 
     elif file:
-        f = spooled_to_bytes_io_if_needed(
-            cast(Union[BinaryIO, SpooledTemporaryFile], file),
```

Comment:
Is there a specific reason to remove the `spooled_to_bytes_io_if_needed()` utility function? IMO, this function is also used for other partitioners.

Reply:
It is, but the typing on it is vexed and fixing that would involve fixing a lot of other files. Basically it's signature should be `IO[bytes] -> IO[bytes]` but it takes `BinaryIO | SpooledTemporaryFile` instead and so it doesn't type-check if you try to pass in a value typed as `IO[bytes]`.

Also its interface is odd because it also accepts `bytes` but then doesn't do anything with it. And what it does do is so simple it's easy to inline.

Anyway, we could undertake that broader fix if you wanted. It looks like it's used in `tsv`, `pdf`, `csv`, and `xml`, so that's not too extensive. Just let me know and I'm happy to do the needful :)

Reply:
And btw, I'm not removing the function, just the call to it here. So it's still available for other partitioners to use as they have been :)